src/app/templates/OperationHistory.tsx

```diff
@@ -17,77 +28,246 @@ import Identicon from "app/atoms/Identicon";
 import HashChip from "app/atoms/HashChip";
 import Money from "app/atoms/Money";
 import { ReactComponent as LayersIcon } from "app/icons/layers.svg";
+import useInfiniteList from "lib/useInfiniteList";
+import FormSecondaryButton from "app/atoms/FormSecondaryButton";
+import {
+  hasAmount,
+  hasReceiver,
+  isTokenTransaction,
+  isTzktTransaction,
+  ThanosHistoricalOperation,
+  ThanosHistoricalTokenTransaction,
+  ThanosHistoricalTzktOperation,
+  ThanosOperation,
+  ThanosPendingOperation,
+} from "lib/transactionHistoryTypings";
 
 const PNDOP_EXPIRE_DELAY = 1000 * 60 * 60 * 24;
 
-interface OperationPreview {
-  hash: string;
-  type: string;
-  receiver: string;
-  volume: number;
-  status: string;
-  time: string;
-  parameters?: any;
-}
-
-interface OperationHistoryProps {
+type OperationHistoryProps = {
   accountPkh: string;
-}
+};
 
 const OperationHistory: React.FC<OperationHistoryProps> = ({ accountPkh }) => {
   const network = useNetwork();
   const { pndOps, removePndOps } = usePendingOperations();
 
-  const fetchOperations = React.useCallback(async () => {
-    try {
-      if (!network.tzStats) return [];
-
-      const { ops } = await getAccountWithOperations(network.tzStats, {
-        pkh: accountPkh,
-        order: "desc",
-        limit: 25,
-        offset: 0,
-      });
-      return ops;
-    } catch (err) {
-      if (err?.origin?.response?.status === 404) {
-        return [];
+  const bcdOperationsGetKey = useCallback(
+    (index: number, previousPageData: BcdPageableTokenTransfers | null) => {
+      if (!previousPageData) {
+        return `bcdOperations,${network.id},${accountPkh}`;
```

Comment:
I've tried replacing this with kosher array of strings. Unfortunately, infinite list hook doesn't make queries in that case